templates/reservation/reservation.py

```diff
@@ -30,9 +32,22 @@ def __init__(self, name, guid=None, data=None):
     def validate(self):
         if not self.data.get('creationTimestamp'):
             self.data['creationTimestamp'] = time.time()
+        if not self.data.get('expiryTimestamp'):
+            self.data['expiryTimestamp'] = time.time()
+
+    def extend(self, duration):
```

Comment:
This is a bit hard to review like that. Can you add a bunch of test to validate the calculation of the extended time is correct ?

Reply:
I moved the functionality under `j.tools.time` and added a test there.
https://github.com/threefoldtech/jumpscale_lib/commit/12ec507e86ab2d0e8593ac64c5a8004d0b018642